# Author: Damyan Ivanov <dmn@debian.org>
# Description: add missing Apache constant imports
# Upstream-Bug: http://rt.cpan.org/Ticket/Display.html?id=41928
# Debian-Bug: http://bugs.debian.org/509625
--- a/lib/SOAP/Transport/HTTP.pm
+++ b/lib/SOAP/Transport/HTTP.pm
@@ -644,7 +644,7 @@ sub new {
         require Apache2::Const;
         require Apache2::RequestUtil;
         require APR::Table;
-        Apache2::Const->import(-compile => 'OK');
+        Apache2::Const->import(-compile => qw(OK HTTP_BAD_REQUEST));
         $self->{'MOD_PERL_VERSION'} = 2;
         $self->{OK} = &Apache2::Const::OK;
     }
@@ -656,7 +656,7 @@ sub new {
         if ($mod_perl::VERSION < 1.99) {
            require Apache;
            require Apache::Constants;
-           Apache::Constants->import('OK');
+           Apache::Constants->import(qw(OK BAD_REQUEST));
            $self->{'MOD_PERL_VERSION'} = 1;
            $self->{OK} = &Apache::Constants::OK;
         }
@@ -664,7 +664,7 @@ sub new {
            require Apache::RequestRec;
            require Apache::RequestIO;
            require Apache::Const;
-           Apache::Const->import(-compile => 'OK');
+           Apache::Const->import(-compile => qw(OK HTTP_BAD_REQUEST));
            $self->{'MOD_PERL_VERSION'} = 1.99;
            $self->{OK} = &Apache::OK;
         }
